### PR TITLE
Added Motorola 68K Grammar

### PR DESCRIPTION
**Pull request by joshumax**
This commit adds Motorola 68K assembly support to linguist.

## Checklist:

- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AX68+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AX68+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [x] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fzerkman%2Flanguage-m68k%2Fmaster%2Fgrammars%2Fm68k.cson&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fjoshumax%2Flinguist%2Fc1168f82b4f5d6d6bb1a8b1581a17b2717b037af%2Fsamples%2FMotorola%252068K%2520Assembly%2Fbasic.X68&code=
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/
  
- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -3065,6 +3065,14 @@ MoonScript:
...
   ace_mode: text
   language_id: 238
+Motorola 68K Assembly:
+  type: programming
+  group: Assembly
+  extensions:
+  - ".X68"
+  tm_scope: source.68k
+  ace_mode: assembly_x86
+  language_id: 432
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: That language_id looks weird. Did you generate it as per the contribution guidelines?

File: samples/Motorola 68K Assembly/basic.X68 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+*-----------------------------------------------------------
```
Comment (pchaigno) [inline samples/Motorola 68K Assembly/basic.X68]: Do you have a more realistic sample file, from a "real" application?

File: grammars.yml [hunk 3]
```diff
@@ -480,6 +480,8 @@ vendor/grammars/language-less:
 - source.css.less
 vendor/grammars/language-maxscript:
 - source.maxscript
+vendor/grammars/language-m68k:
```
Comment (pchaigno) [inline grammars.yml]: How did you add the grammar? With the provided script?

**Comment by joshumax** [comment]
@pchaigngo Thank you for your feedback! These problems have been fixed.

File: lib/linguist/languages.yml [hunk 4]
```diff
@@ -3065,6 +3065,14 @@ MoonScript:
...
   ace_mode: text
   language_id: 238
+Motorola 68K Assembly:
+  type: programming
+  group: Assembly
+  extensions:
+  - ".X68"
+  tm_scope: source.m68k
+  ace_mode: assembly_x86
+  language_id: 477582706
```
Comment (lildude) [inline lib/linguist/languages.yml]: I note you've not picked a colour for this language. That's not a problem as we'll fall back to the default of `#cccccc` but I wanted to check if this is intentional.

Comment (pchaigno) [reply lib/linguist/languages.yml]: It's part of a group so it won't get a color anyway, no?

Comment (lildude) [reply lib/linguist/languages.yml]: Doh!! Of course. I completely missed that. Thanks @pchaigno

**Comment by lildude** [comment]
@joshumax can you please resolve the merge conflict.

**Comment by joshumax** [comment]
@lildude @pchaigno The merge conflict has been resolved.

**Comment by lildude** [comment]
Thanks. The test failure needs to be addressed now:

```
  1) Failure:
TestGrammars#test_readme_file_is_in_sync [/home/travis/build/github/linguist/test/test_grammars.rb:42]:
Grammar list is out-of-date. Run `script/list-grammars`.
--- expected
+++ actual
@@ -242,11 +242,6 @@
 - **Monkey:** [gingerbeardman/monkey.tmbundle](https://github.com/gingerbeardman/monkey.tmbundle)
 - **MoonScript:** [leafo/moonscript-tmbundle](https://github.com/leafo/moonscript-tmbundle)
 - **Motorola 68K Assembly:** [zerkman/language-m68k](https://github.com/zerkman/language-m68k)
-- **MQL4:** [mqsoft/MQL5-sublime](https://github.com/mqsoft/MQL5-sublime)
-- **MQL5:** [mqsoft/MQL5-sublime](https://github.com/mqsoft/MQL5-sublime)
-- **MTML:** [atom/language-html](https://github.com/atom/language-html)
-- **mupad:** [ccreutzig/sublime-MuPAD](https://github.com/ccreutzig/sublime-MuPAD)
-- **nanorc:** [Alhadis/language-etc](https://github.com/Alhadis/language-etc)
 - **NCL:** [rpavlick/language-ncl](https://github.com/rpavlick/language-ncl)
 - **NSIS:** [github-linguist/NSIS](https://github.com/github-linguist/NSIS)
 - **Nearley:** [Hardmath123/sublime-nearley](https://github.com/Hardmath123/sublime-nearley)
```

**Comment by lildude** [comment]
@joshumax nudge

**Comment by joshumax** [comment]
@lildude Whoops, looks like I missed the few messages above and it went stale. I'll push it up!

File: vendor/README.md [hunk 5]
```diff
@@ -247,6 +247,12 @@ This is a list of grammars that Linguist selects to provide syntax highlighting
 - **Modula-3:** [newgrammars/m3](https://github.com/newgrammars/m3)
 - **Monkey:** [gingerbeardman/monkey.tmbundle](https://github.com/gingerbeardman/monkey.tmbundle)
 - **MoonScript:** [leafo/moonscript-tmbundle](https://github.com/leafo/moonscript-tmbundle)
+- **Motorola 68K Assembly:** [zerkman/language-m68k](https://github.com/zerkman/language-m68k)
+- **MQL4:** [mqsoft/MQL5-sublime](https://github.com/mqsoft/MQL5-sublime)
```
Comment (lildude) [inline vendor/README.md]: This line and the 4 below it are what is causing the test to fail. They need to be removed from this PR.

**Comment by Alhadis** [comment]
@joshumax Just run `script/list-grammars` locally to regenerate the grammars list.

**Pull request merged**